File: vue-project/src/api/talentMarketController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utils/request'

/** 搜索人才 POST /talent-market/search */
export async function searchTalents(body: API.TalentSearchRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponsePageTalentVO>('/talent-market/search', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 预估高级搜索消耗的积分 POST /talent-market/search/cost-preview */
export async function getAdvancedSearchCostPreview(body: API.TalentSearchRequest, options?: { [key: string]: any }) {
  return request<API.BaseResponseBigDecimal>('/talent-market/search/cost-preview', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 获取人才详情 GET /talent-market/detail/${employeeId} */
export async function getTalentDetail(
  params: { employeeId: string },
  options?: { [key: string]: any },
) {
  const { employeeId } = params
  return request<API.BaseResponseTalentDetailVO>(`/talent-market/detail/${employeeId}`, {
    method: 'GET',
    ...(options || {}),
  })
}

/** 解锁评价 POST /talent-market/unlock */
export async function unlockEvaluation(
  body: API.EvaluationUnlockRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBigDecimal>('/talent-market/unlock', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 批量解锁评价 POST /talent-market/unlock/batch */
export async function batchUnlockEvaluations(
  body: API.EvaluationUnlockRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBigDecimal>('/talent-market/unlock/batch', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 收藏人才 POST /talent-market/bookmark */
export async function bookmarkTalent(
  body: API.TalentBookmarkRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseLong>('/talent-market/bookmark', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 取消收藏人才 DELETE /talent-market/bookmark/${employeeId} */
export async function unbookmarkTalent(
  params: { employeeId: string },
  options?: { [key: string]: any },
) {
  const { employeeId } = params
  return request<API.BaseResponseBoolean>(`/talent-market/bookmark/${employeeId}`, {
    method: 'DELETE',
    ...(options || {}),
  })
}

/** 获取收藏的人才列表 GET /talent-market/bookmarks */
export async function getBookmarkedTalents(
  params: { pageNum?: number; pageSize?: number },
  options?: { [key: string]: any },
) {
  return request<API.BaseResponsePageTalentVO>('/talent-market/bookmarks', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 获取解锁价格配置 GET /talent-market/unlock-prices */
export async function getUnlockPriceConfigs(options?: { [key: string]: any }) {
  return request<API.BaseResponseListUnlockPriceConfigVO>('/talent-market/unlock-prices', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 获取所有评价标签 GET /talent-market/tags */
export async function getAllTags(options?: { [key: string]: any }) {
  return request<API.BaseResponseListEvaluationTagVO>('/talent-market/tags', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 检查权限 GET /talent-market/check-permission */
export async function checkPermission(options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean>('/talent-market/check-permission', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 记录人才浏览 POST /talent-market/view/record */
export async function recordView(
  body: API.TalentViewLogRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseLong>('/talent-market/view/record', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 更新浏览时长 POST /talent-market/view/duration */
export async function updateViewDuration(
  params: { viewLogId: string; viewDuration: number },
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBoolean>('/talent-market/view/duration', {
    method: 'POST',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 获取浏览历史 GET /talent-market/view/history */
export async function getViewHistory(
  params: { pageNum?: number; pageSize?: number },
  options?: { [key: string]: any },
) {
  return request<API.BaseResponsePageTalentViewLogVO>('/talent-market/view/history', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 获取浏览统计 GET /talent-market/view/statistics */
export async function getViewStatistics(options?: { [key: string]: any }) {
  return request<API.BaseResponseViewStatisticsVO>('/talent-market/view/statistics', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 清除浏览记录 DELETE /talent-market/view/history */
export async function clearViewHistory(
  params?: { employeeId?: string },
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseInt>('/talent-market/view/history', {
    method: 'DELETE',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 保存企业偏好 POST /talent-market/preference */
export async function savePreference(
  body: API.CompanyPreferenceRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseLong>('/talent-market/preference', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 获取企业偏好 GET /talent-market/preference */
export async function getPreference(options?: { [key: string]: any }) {
  return request<API.BaseResponseCompanyPreferenceVO>('/talent-market/preference', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 获取推荐人才 GET /talent-market/recommend */
export async function getRecommendedTalents(
  params: { pageNum?: number; pageSize?: number },
  options?: { [key: string]: any },
) {
  return request<API.BaseResponsePageTalentRecommendVO>('/talent-market/recommend', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 获取相似人才 GET /talent-market/recommend/similar/${employeeId} */
export async function getSimilarTalents(
  params: { employeeId: string; limit?: number },
  options?: { [key: string]: any },
) {
  const { employeeId, ...queryParams } = params
  return request<API.BaseResponseListTalentRecommendVO>(
    `/talent-market/recommend/similar/${employeeId}`,
    {
      method: 'GET',
      params: {
        ...queryParams,
      },
      ...(options || {}),
    },
  )
}

/** 对比人才 POST /talent-market/compare */
export async function compareTalents(
  body: API.TalentCompareRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseTalentCompareVO>('/talent-market/compare', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

File: src/main/java/com/crossorgtalentmanager/controller/TalentMarketController.java
package com.crossorgtalentmanager.controller;

import com.crossorgtalentmanager.annotation.AuthCheck;
import com.crossorgtalentmanager.common.BaseResponse;
import com.crossorgtalentmanager.common.ResultUtils;
import com.crossorgtalentmanager.constant.UserConstant;
import com.crossorgtalentmanager.exception.ErrorCode;
import com.crossorgtalentmanager.exception.ThrowUtils;
import com.crossorgtalentmanager.model.dto.talentmarket.*;
import com.crossorgtalentmanager.model.entity.User;
import com.crossorgtalentmanager.model.vo.*;
import com.crossorgtalentmanager.ratelimiter.annotation.RateLimit;
import com.crossorgtalentmanager.ratelimiter.enums.RateLimitType;
import com.crossorgtalentmanager.service.TalentMarketService;
import com.crossorgtalentmanager.service.UserService;
import com.mybatisflex.core.paginate.Page;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.*;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * 人才市场 控制层
 *
 * @author y
 */
@RestController
@RequestMapping("/talent-market")
public class TalentMarketController {

    @Resource
    private TalentMarketService talentMarketService;

    @Resource
    private UserService userService;

    /**
     * 搜索人才（分页）
     * 支持多条件筛选：职位、评分、标签、评价内容关键词等
     *
     * @param request     搜索请求
     * @param httpRequest HTTP请求
     * @return 人才列表分页结果
     */
    @PostMapping("/search")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Page<TalentVO>> searchTalents(@RequestBody TalentSearchRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        Page<TalentVO> page = talentMarketService.searchTalents(request, loginUser);
        return ResultUtils.success(page);
    }

    /**
     * 预估高级搜索消耗的积分
     * 用于在搜索前提示用户将消耗多少积分
     *
     * @param request 搜索请求
     * @return 预估消耗的积分
     */
    @PostMapping("/search/cost-preview")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<BigDecimal> getAdvancedSearchCostPreview(@RequestBody TalentSearchRequest request) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        BigDecimal cost = talentMarketService.calculateAdvancedSearchCostPreview(request);
        return ResultUtils.success(cost);
    }

    /**
     * 获取人才详情
     * 包含工作经历、评价（带解锁状态）、标签统计等
     *
     * @param employeeId  员工ID
     * @param httpRequest HTTP请求
     * @return 人才详情
     */
    @GetMapping("/detail/{employeeId}")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<TalentDetailVO> getTalentDetail(@PathVariable Long employeeId,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(employeeId == null || employeeId <= 0, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        TalentDetailVO detail = talentMarketService.getTalentDetail(employeeId, loginUser);
        return ResultUtils.success(detail);
    }

    /**
     * 解锁评价（消耗积分）
     *
     * @param request     解锁请求
     * @param httpRequest HTTP请求
     * @return 消耗的积分
     */
    @PostMapping("/unlock")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<BigDecimal> unlockEvaluation(@RequestBody EvaluationUnlockRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        BigDecimal cost = talentMarketService.unlockEvaluation(request, loginUser);
        return ResultUtils.success(cost);
    }

    /**
     * 批量解锁评价（消耗积分）
     *
     * @param request     解锁请求（包含评价ID列表）
     * @param httpRequest HTTP请求
     * @return 消耗的总积分
     */
    @PostMapping("/unlock/batch")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<BigDecimal> batchUnlockEvaluations(@RequestBody EvaluationUnlockRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        BigDecimal cost = talentMarketService.batchUnlockEvaluations(request, loginUser);
        return ResultUtils.success(cost);
    }

    /**
     * 收藏人才
     *
     * @param request     收藏请求
     * @param httpRequest HTTP请求
     * @return 收藏ID
     */
    @PostMapping("/bookmark")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Long> bookmarkTalent(@RequestBody TalentBookmarkRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        Long bookmarkId = talentMarketService.bookmarkTalent(request, loginUser);
        return ResultUtils.success(bookmarkId);
    }

    /**
     * 取消收藏人才
     *
     * @param employeeId  员工ID
     * @param httpRequest HTTP请求
     * @return 是否成功
     */
    @DeleteMapping("/bookmark/{employeeId}")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Boolean> unbookmarkTalent(@PathVariable Long employeeId,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(employeeId == null || employeeId <= 0, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        boolean result = talentMarketService.unbookmarkTalent(employeeId, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 获取收藏的人才列表（分页）
     *
     * @param pageNum     页码
     * @param pageSize    每页大小
     * @param httpRequest HTTP请求
     * @return 收藏的人才列表
     */
    @GetMapping("/bookmarks")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Page<TalentVO>> getBookmarkedTalents(
            @RequestParam(defaultValue = "1") long pageNum,
            @RequestParam(defaultValue = "10") long pageSize,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        Page<TalentVO> page = talentMarketService.getBookmarkedTalents(pageNum, pageSize, loginUser);
        return ResultUtils.success(page);
    }

    /**
     * 获取解锁价格配置
     *
     * @return 价格配置列表
     */
    @GetMapping("/unlock-prices")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<List<UnlockPriceConfigVO>> getUnlockPriceConfigs() {
        List<UnlockPriceConfigVO> configs = talentMarketService.getUnlockPriceConfigs();
        return ResultUtils.success(configs);
    }

    /**
     * 获取所有评价标签（用于筛选）
     *
     * @return 标签列表
     */
    @GetMapping("/tags")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<List<EvaluationTagVO>> getAllTags() {
        List<EvaluationTagVO> tags = talentMarketService.getAllTags();
        return ResultUtils.success(tags);
    }

    /**
     * 检查当前用户是否有权限访问人才市场
     *
     * @param httpRequest HTTP请求
     * @return 是否有权限
     */
    @GetMapping("/check-permission")
    public BaseResponse<Boolean> checkPermission(HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        boolean hasPermission = talentMarketService.hasAccessPermission(loginUser);
        return ResultUtils.success(hasPermission);
    }

    // ==================== 人才浏览记录相关 ====================

    /**
     * 记录人才浏览
     *
     * @param request     浏览记录请求
     * @param httpRequest HTTP请求
     * @return 浏览记录ID
     */
    @PostMapping("/view/record")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Long> recordView(@RequestBody TalentViewLogRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        Long viewLogId = talentMarketService.recordView(request, loginUser);
        return ResultUtils.success(viewLogId);
    }

    /**
     * 更新浏览时长
     *
     * @param viewLogId    浏览记录ID
     * @param viewDuration 浏览时长（秒）
     * @param httpRequest  HTTP请求
     * @return 是否成功
     */
    @PostMapping("/view/duration")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Boolean> updateViewDuration(@RequestParam Long viewLogId,
            @RequestParam Integer viewDuration,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        boolean result = talentMarketService.updateViewDuration(viewLogId, viewDuration, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 获取浏览历史（分页）
     *
     * @param pageNum     页码
     * @param pageSize    每页大小
     * @param httpRequest HTTP请求
     * @return 浏览历史列表
     */
    @GetMapping("/view/history")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Page<TalentViewLogVO>> getViewHistory(
            @RequestParam(defaultValue = "1") long pageNum,
            @RequestParam(defaultValue = "10") long pageSize,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        Page<TalentViewLogVO> page = talentMarketService.getViewHistory(pageNum, pageSize, loginUser);
        return ResultUtils.success(page);
    }

    /**
     * 获取浏览统计数据
     *
     * @param httpRequest HTTP请求
     * @return 统计数据
     */
    @GetMapping("/view/statistics")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<ViewStatisticsVO> getViewStatistics(HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        ViewStatisticsVO stats = talentMarketService.getViewStatistics(loginUser);
        return ResultUtils.success(stats);
    }

    /**
     * 清除浏览记录
     *
     * @param employeeId  员工ID（可选，为空则清除全部）
     * @param httpRequest HTTP请求
     * @return 清除的记录数
     */
    @DeleteMapping("/view/history")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Integer> clearViewHistory(
            @RequestParam(required = false) Long employeeId,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        int count = talentMarketService.clearViewHistory(employeeId, loginUser);
        return ResultUtils.success(count);
    }

    // ==================== 企业招聘偏好相关 ====================

    /**
     * 保存或更新企业招聘偏好
     *
     * @param request     偏好设置请求
     * @param httpRequest HTTP请求
     * @return 偏好ID
     */
    @PostMapping("/preference")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Long> savePreference(@RequestBody CompanyPreferenceRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        Long preferenceId = talentMarketService.savePreference(request, loginUser);
        return ResultUtils.success(preferenceId);
    }

    /**
     * 获取企业招聘偏好
     *
     * @param httpRequest HTTP请求
     * @return 偏好设置
     */
    @GetMapping("/preference")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<CompanyPreferenceVO> getPreference(HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        CompanyPreferenceVO preference = talentMarketService.getPreference(loginUser);
        return ResultUtils.success(preference);
    }

    // ==================== 人才推荐相关 ====================

    /**
     * 获取推荐人才列表（基于企业偏好和浏览历史）
     *
     * @param pageNum     页码
     * @param pageSize    每页大小
     * @param httpRequest HTTP请求
     * @return 推荐人才列表
     */
    @GetMapping("/recommend")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Page<TalentRecommendVO>> getRecommendedTalents(
            @RequestParam(defaultValue = "1") long pageNum,
            @RequestParam(defaultValue = "10") long pageSize,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        Page<TalentRecommendVO> page = talentMarketService.getRecommendedTalents(pageNum, pageSize, loginUser);
        return ResultUtils.success(page);
    }

    /**
     * 获取相似人才推荐（基于指定员工）
     *
     * @param employeeId  参考员工ID
     * @param limit       推荐数量
     * @param httpRequest HTTP请求
     * @return 相似人才列表
     */
    @GetMapping("/recommend/similar/{employeeId}")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<List<TalentRecommendVO>> getSimilarTalents(
            @PathVariable Long employeeId,
            @RequestParam(defaultValue = "5") int limit,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(employeeId == null || employeeId <= 0, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        List<TalentRecommendVO> talents = talentMarketService.getSimilarTalents(employeeId, limit, loginUser);
        return ResultUtils.success(talents);
    }

    // ==================== 人才对比相关 ====================

    /**
     * 对比多个人才
     *
     * @param request     对比请求（包含员工ID列表，2-5个）
     * @param httpRequest HTTP请求
     * @return 对比结果
     */
    @PostMapping("/compare")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    @RateLimit(limitType = RateLimitType.USER, rate = 2, rateInterval = 60, message = "提交分析任务过于频繁，请稍后再试")
    public BaseResponse<TalentCompareVO> compareTalents(@RequestBody TalentCompareRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        TalentCompareVO compareResult = talentMarketService.compareTalents(request, loginUser);
        return ResultUtils.success(compareResult);
    }

    /**
     * 查询AI分析结果
     *
     * @param taskId      任务ID
     * @param httpRequest HTTP请求
     * @return AI分析结果
     */
    @GetMapping("/compare/ai-result")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    @RateLimit(limitType = RateLimitType.USER, rate = 120, rateInterval = 60, message = "请求过于频繁，请稍后再试")
    public BaseResponse<Map<String, Object>> getAiAnalysisResult(@RequestParam String taskId,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(taskId == null || taskId.trim().isEmpty(), ErrorCode.PARAMS_ERROR, "任务ID不能为空");
        User loginUser = userService.getLoginUser(httpRequest);

        Map<String, Object> result = new HashMap<>();
        String status = talentMarketService.getAiAnalysisTaskStatus(taskId);
        String aiResult = talentMarketService.getAiAnalysisTaskResult(taskId);
        String error = talentMarketService.getAiAnalysisTaskError(taskId);

        result.put("status", status);
        result.put("result", aiResult);
        result.put("error", error);

        // 如果AI分析完成，更新对比记录中的AI分析结果
        if ("completed".equals(status) && aiResult != null && !aiResult.trim().isEmpty()) {
            talentMarketService.updateCompareRecordAiResult(taskId, aiResult, loginUser);
        }

        return ResultUtils.success(result);
    }

    /**
     * 查询历史对比记录
     *
     * @param pageNum     页码
     * @param pageSize    每页大小
     * @param httpRequest HTTP请求
     * @return 历史对比记录列表
     */
    @GetMapping("/compare/history")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<Page<TalentCompareRecordVO>> getCompareHistory(
            @RequestParam(defaultValue = "1") long pageNum,
            @RequestParam(defaultValue = "10") long pageSize,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        Page<TalentCompareRecordVO> page = talentMarketService.getCompareHistory(pageNum, pageSize, loginUser);
        return ResultUtils.success(page);
    }

    /**
     * 根据记录ID获取历史对比记录详情
     *
     * @param recordId    记录ID
     * @param httpRequest HTTP请求
     * @return 历史对比记录详情
     */
    @GetMapping("/compare/history/{recordId}")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<TalentCompareRecordVO> getCompareHistoryById(
            @PathVariable Long recordId,
            HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        TalentCompareRecordVO record = talentMarketService.getCompareHistoryById(recordId, loginUser);
        return ResultUtils.success(record);
    }

    /**
     * 检查是否存在相同的对比记录
     *
     * @param request     对比请求（包含员工ID列表）
     * @param httpRequest HTTP请求
     * @return 如果存在返回记录信息，否则返回null
     */
    @PostMapping("/compare/check-existing")
    @AuthCheck(mustRole = UserConstant.HR_ROLE)
    public BaseResponse<TalentCompareRecordVO> checkExistingCompare(@RequestBody TalentCompareRequest request,
            HttpServletRequest httpRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpRequest);
        TalentCompareRecordVO record = talentMarketService.checkExistingCompare(request, loginUser);
        return ResultUtils.success(record);
    }
}
